flink-metrics/flink-metrics-prometheus/src/test/java/org/apache/flink/metrics/prometheus/PrometheusReporterTest.java

```diff
@@ -160,6 +151,43 @@ public void invalidCharactersAreReplacedWithUnderscore() {
 		assertThat(PrometheusReporter.replaceInvalidChars("a,=;:?'b,=;:?'c"), equalTo("a___:__b___:__c"));
 	}
 
+	@Test
+	public void registeringSameMetricTwiceDoesNotThrowException() {
+		Counter counter = new SimpleCounter();
+		counter.inc();
+		String counterName = "testCounter";
+		final FrontMetricGroup<TaskManagerMetricGroup> group = new FrontMetricGroup<>(0, new TaskManagerMetricGroup(registry, HOST_NAME, TASK_MANAGER));
+
+		reporter.notifyOfAddedMetric(counter, counterName, group);
+		reporter.notifyOfAddedMetric(counter, counterName, group);
+	}
+
+	@Test
+	public void cannotStartTwoReportersOnSamePort() {
+		final MetricRegistry fixedPort1 = new MetricRegistry(MetricRegistryConfiguration.fromConfiguration(createConfigWithOneReporter("test1", "12345")));
+		final MetricRegistry fixedPort2 = new MetricRegistry(MetricRegistryConfiguration.fromConfiguration(createConfigWithOneReporter("test2", "12345")));
+		assertThat(fixedPort1.getReporters(), hasSize(1));
+		assertThat(fixedPort2.getReporters(), hasSize(0));
+	}
+
+	@Test
+	public void canStartTwoReportersWhenUsingPortRange() {
+		final MetricRegistry portRange1 = new MetricRegistry(MetricRegistryConfiguration.fromConfiguration(createConfigWithOneReporter("test1", "9249-9252")));
+		final MetricRegistry portRange2 = new MetricRegistry(MetricRegistryConfiguration.fromConfiguration(createConfigWithOneReporter("test2", "9249-9252")));
+		assertThat(portRange1.getReporters(), hasSize(1));
+		assertThat(portRange2.getReporters(), hasSize(1));
+	}
+
+	@Test
+	public void cannotStartThreeReportersWhenPortRangeIsTooSmall() {
+		final MetricRegistry smallPortRange1 = new MetricRegistry(MetricRegistryConfiguration.fromConfiguration(createConfigWithOneReporter("test1", "9253-9254")));
```

Comment:
You should call `MetricRegistry#shutdown()` when you no longer need it.

Reply:
done